src/sidebar/services/frame-sync.js

```diff
@@ -181,19 +181,33 @@ export class FrameSyncService {
   _connectGuest(port) {
     /** @type {PortRPC<GuestToSidebarEvent, SidebarToGuestEvent>} */
     const guestRPC = new PortRPC();
+    this._guestRPC.push(guestRPC);
+
+    /** @type {string|null} */
+    let frameIdentifier;
 
     // Update document metadata for this guest. We currently assume that the
     // guest will make this call once after it connects. To handle updates
     // to the document, we'll need to change `connectFrame` to update rather than
     // add to the frame list.
     guestRPC.on('documentInfoChanged', info => {
+      frameIdentifier = info.frameIdentifier;
       this._store.connectFrame({
         id: info.frameIdentifier,
         metadata: info.metadata,
         uri: info.uri,
       });
     });
 
+    guestRPC.on('frameDestroyed', () => {
+      const frame = this._store.frames().find(f => f.id === frameIdentifier);
```

Comment:
It feels that `this._store.frames` could be a `Map` instead of an array.

Reply:
The reason that `frames` returns an array currently is because it returns data from a Redux store which only contains plain objects and arrays that are easily made immutable. It might make sense to add support for a lightweight immutable maps/sets in the store in future.